server/src/main/java/org/apache/iotdb/db/engine/compaction/CompactionWorker.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iotdb.db.engine.compaction;
+
+import org.apache.iotdb.db.engine.compaction.constant.CompactionTaskStatus;
+import org.apache.iotdb.db.engine.compaction.task.AbstractCompactionTask;
+import org.apache.iotdb.db.engine.compaction.task.CompactionTaskSummary;
+import org.apache.iotdb.db.service.metrics.recorder.CompactionMetricsRecorder;
+import org.apache.iotdb.db.utils.datastructure.FixedPriorityBlockingQueue;
+
+import org.jetbrains.annotations.NotNull;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+public class CompactionWorker implements Runnable {
+  private static final Logger log = LoggerFactory.getLogger("COMPACTION");
+  private final int threadId;
+  private final FixedPriorityBlockingQueue<AbstractCompactionTask> compactionTaskQueue;
+
+  public CompactionWorker(
+      int threadId, FixedPriorityBlockingQueue<AbstractCompactionTask> compactionTaskQueue) {
+    this.threadId = threadId;
+    this.compactionTaskQueue = compactionTaskQueue;
+  }
+
+  @Override
+  public void run() {
+    while (!Thread.currentThread().isInterrupted()) {
+      try {
+        AbstractCompactionTask task = compactionTaskQueue.take();
+        // add metrics
+        CompactionMetricsRecorder.recordTaskInfo(
+            task, CompactionTaskStatus.POLL_FROM_QUEUE, compactionTaskQueue.size());
+        if (task != null && task.checkValidAndSetMerging()) {
+          CompactionTaskSummary summary = task.getSummary();
+          CompactionTaskFuture future = new CompactionTaskFuture(summary);
+          CompactionTaskManager.getInstance().recordTask(task, future);
+          task.call();
```

Comment:
As  AbstractCompactionTask is not submitted to some ThreadPool, it is no longer to implement the Callable interface.

Reply:
resolve